docs/kb/semgrep-ci/scan-monorepo-in-parts.md

```diff
@@ -0,0 +1,54 @@
+2. Logically splitting up the monorepo into multiple components to help with managing findings
+
+## How to configure Semgrep in CI to split up a monorepo
+
+Typically when scanning any repo with Semgrep in CI, the base command is `semgrep ci`. Of course, you need to have an auth token set and perhaps other environment variables configured depending on what SCM and CI provider you are using, but this article will assume you are already familiar with those settings.
+
+To split up your monorepo, you need to first decide *how* you want to logically split up the code. For example, if your monorepo has four main modules and their paths are:
+
+    /src/moduleA
+    /src/moduleB
+    /src/moduleC
+    /src/moduleD
+
+Then it would logically make sense to split this into four separate scans in CI, one for each module.
+
+To actually achieve this, you would use the `--include` flag ([see CLI reference](/docs/cli-reference/)) to only scan files in each module's code path:
+
+    semgrep ci --include=/src/moduleA/*
+
+Now, Semgrep is only scanning files under that path and the entire CI run will take less time, since less code is being scanned. Of course, you will now have to run multiple CI runs for the remaining three modules to get full monorepo coverage, but you have more flexibility on how and when you perform those runs.
+
+Now that you've successfully configured your monorepo to be scanned in parts, you also have to configure the findings from each part or module to show up as their own project in Semgrep Cloud Platform (SCP).
+
+Typically, findings from a single repository are associated to a single project in SCP. However you can change this by manually setting the `SEMGREP_REPO_NAME` environment variable ([see CI environment variables reference](/docs/semgrep-ci/ci-environment-variables/#semgrep_repo_name)).
+
+For example, if your monorepo is located at `https://github.com/sina/monorepo` the `SEMGREP_REPO_NAME` would typically be set to `sina/monorepo`. So to split it up into the four modules we referenced above, for each CI run of a module we need to manually set it like so before running Semgrep:
+
+    export SEMGREP_REPO_NAME="sina/monorepo/moduleA"
+
+And then running Semgrep as demonstrated above:
+
+    semgrep ci --include=/src/moduleA/*
+
+Now, the findings from this CI run will show up in their own project in SCP named `sina/monorepo/moduleA`. This is not only necessary when splitting up a monorepo but also helpful in terms of organizing findings into separate projects so that devs and security engineers can have a clearer understanding of which findings pertain to the module that they are actually responsible for.
+
+:::info
+Due to how we determine hyperlinks for findings in SCP, changing the `SEMGREP_REPO_NAME` to anything other than the actual `<org>/<repo name>` format may adversely affect and break hyperlinks. This is a tradeoff of splitting up one single repo into multiple projects.
+:::
```

Comment:
I'd move this up as well, likely keeping it as an info box wherever you first introduce the SEMGREP_REPO_NAME usage. I'd write it a bit differently though:

> Changing the `SEMGREP_REPO_NAME` value in a scan so that it does not match the repo's `<org>/<repo name>` value on the SCM may cause issues with code hyperlinks in Semgrep Cloud Platform. This is a necessary tradeoff when splitting up a repo into multiple projects.